### PR TITLE
Fix/force paths out es home

### LINKED ISSUE #186: MSI installer 6.3.0 - Config files deleted after update
**Found in version**
- 6.3.0-SNAPSHOT

**Steps to reproduce**
- Install elasticsearch version 6.2.4 with the MSI installer and select the checkbox to use the install directory (e.g. `C:\Elasticsearch\6.2.4\`) for config, data and logs as well
- Update to 6.3.0 with the MSI installer

**Expected result**
- The config files are updated if needed but are still in the same (old) directory

**Actual result**
- The files `elasticsearch.yml` and `jvm.options` are deleted
- The elasticsearch service is running with updated configuration but a restart of the service results in an error (so it seems the config files have been updated ok and then been deleted after 6.3.0 service start)

### PR DESCRIPTION
**Pull request by Mpdreamz**
We used to support upgrading from zip files and thus this option made sense. However since quite a while with the way our Wix engine executes the install sequence we no longer support this.

Putting conf/data/path in the same folder as the installation is not recommended so lets not expose this option. With this PR it will even be a validation error if conf/data/path is a sub folder of the
installation.

image

Further more if we see that `ES_PATH_CONF` is a subfolder of `ES_HOME` at the start of the installation we block the installation from commencing:

image

This should fix #186 when doing an upgrade from a previous installation that had opted in to placed the special folders inside the installation directory.

Finally if we spot an `ES_HOME` but no previous installation we also prohibit the installation from beginning:

image

This should make it explicit that we do not support installations over previously zip installed Elasticsearch instances.

### CODE REVIEW

File: src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs [hunk 1]
```diff
@@ -142,57 +132,26 @@ public void SetDefaultLocations()
...
-		// ReSharper disable InconsistentNaming
-		// ReSharper disable ArrangeTypeMemberModifiers
-		readonly ObservableAsPropertyHelper<bool> configureAllLocations;
-		public bool ConfigureAllLocations => configureAllLocations.Value;
-
 		readonly ObservableAsPropertyHelper<string> elasticsearchLog;
 		public string ElasticsearchLog => elasticsearchLog.Value;
 
-		bool placeWriteableLocationsInSamePath;
-		[Argument(nameof(PlaceWritableLocationsInSamePath))]
```
Comment (russcam) [inline src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs]: Need to update documentation in README and on elastic.co with this change

File: src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs [hunk 2]
```diff
@@ -142,57 +132,26 @@ public void SetDefaultLocations()
 			this._refreshing = false;
 		}
 
-		public bool SamePathAs(string pathA, string pathB)
+		private bool SamePathAs(string pathA, string pathB)
```
Comment (russcam) [inline src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs]: Move to `StringPathExtensions`?

File: src/Elastic.Configuration/Extensions/StringPathExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+		/// Returns true if <paramref name="path"/> starts with the path <paramref name="baseDirPath"/>.
+		/// The comparison is case-insensitive, handles / and \ slashes as folder separators and
+		/// only matches if the base dir folder name is matched exactly ("c:\foobar\file.txt" is not a sub path of "c:\foo").
+		/// </summary>
+		public static bool IsSubPathOf(this string path, string baseDirPath)
+		{
+			var normalizedPath = Path.GetFullPath(path.Replace('/', '\\')
+				.WithEnding("\\"));
+
+			var normalizedBaseDirPath = Path.GetFullPath(baseDirPath.Replace('/', '\\')
```
[the remark this answers is not in the document]
Comment (Mpdreamz) [reply src/Elastic.Configuration/Extensions/StringPathExtensions.cs]: 👍

File: src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs (hunk 2 above)
Comment (Mpdreamz) [reply src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs]: 👍

File: src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs (hunk 1 above)
Comment (Mpdreamz) [reply src/Installer/Elastic.Installer.Domain/Model/Elasticsearch/Locations/LocationsModel.cs]: This was undocumented in the `README` as far as i could tell :)

**Pull request merged**